cpp/include/functions.h

```diff
@@ -15,65 +15,189 @@
  * @Param[in] *source_indices        This gdf_column of size E (number of edges) contains the index of the source for each edge.
- *                                   Indices must be in the range [0, V-1]. 
- * @Param[in] *destination_indices   This gdf_column of size E (number of edges) contains the index of the destination for each edge. 
  *                                   Indices must be in the range [0, V-1].
- * @Param[in] *edge_data (optional)  This pointer can be null_ptr. If not, this gdf_column of size E (number of edges) contains the weiht for each edge. 
+ * @Param[in] *destination_indices   This gdf_column of size E (number of edges) contains the index of the destination for each edge.
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in] *edge_data (optional)  This pointer can be nullptr. If not, this gdf_column of size E (number of edges) contains the weiht for each edge.
  *                                   The type expected to be floating point.
- * 
+ *
  * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added edge list (edge_data is optional).
- * 
- * @Returns                          GDF_SUCCESS upon successful completion. 
+ *
+ * @Returns                          GDF_SUCCESS upon successful completion.
  */
 /* ----------------------------------------------------------------------------*/
-gdf_error gdf_edge_list_view(gdf_graph *graph, 
+gdf_error gdf_edge_list_view(gdf_graph *graph,
                              const gdf_column *source_indices,
                              const gdf_column *destination_indices,
                              const gdf_column *edge_data);
 
-gdf_error gdf_adj_list_view (gdf_graph *graph, 
+/**
+ * @Synopsis   Wrap existing gdf columns representing adjacency lists in a gdf_graph.
+ *             cuGRAPH does not own the memory used to represent this graph. This function does not allocate memory.
+ * @Param[in] *offsets               This gdf_column of size V+1 (V is number of vertices) contains the offset of adjacency lists of every vertex.
+ *                                   Offsets must be in the range [0, E] (number of edges).
+ * @Param[in] *indices               This gdf_column of size E contains the index of the destination for each edge.
+ *                                   Indices must be in the range [0, V-1].
+ * @Param[in] *edge_data (optional)  This pointer can be nullptr. If not, this gdf_column of size E (number of edges) contains the weiht for each edge.
+ *                                   The type expected to be floating point.
+ *
+ * @Param[out] *graph                cuGRAPH graph descriptor containing the newly added adjacency list (edge_data is optional).
+ *
+ * @Returns                          GDF_SUCCESS upon successful completion.
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_adj_list_view (gdf_graph *graph,
```

Comment:
we could add a sentence saying that “This function does not delete any existing data in the cuGRAPH graph descriptor ”  for all add and view functions

Reply:
done